include/alpaka/workdiv/WorkDivHelpers.hpp

```diff
@@ -381,4 +381,17 @@ namespace alpaka
     {
         return isValidWorkDiv(getAccDevProps<TAcc>(dev), workDiv);
     }
+
+    //! \tparam TDim The dimensionality of the accelerator device properties.
+    //! \tparam TIdx The idx type of the accelerator device properties.
+    //! \return Returns the work division; namely the WorkDivMembers instance.
+    template<typename TIdx, typename TDim>
+    auto createWorkDiv(
+        alpaka::Vec<TDim, TIdx> const& gridBlockExtent,
+        alpaka::Vec<TDim, TIdx> const& blockThreadExtent,
+        alpaka::Vec<TDim, TIdx> const& elemExtent) -> alpaka::WorkDivMembers<TDim, TIdx>
+    {
+        return alpaka::WorkDivMembers<TDim, TIdx>(gridBlockExtent, blockThreadExtent, elemExtent);
+    }
+
```

Comment:
Why can't we just add such a constructor to `WorkDivMembers` and a deduction guide if necessary?

Reply:
Yes, but I wanted to use "create..." since I will add new similar functions. Dependency injection seems cleaner for conversions from stl container types and initializer lists.

By the way; I think you mean _deduction guides_ for constructors as the one below, since they don't have a return type?:

```
template<class A, class B>
struct Agg
{
    A a;
    B b;
};
// implicitly-generated guides are formed from default, copy, and move constructors
 
template<class A, class B>
Agg(A a, B b) -> Agg<A, B>;
```

Reply:
Currently we cannot use stl containers, argument types are constrained to alpaka vector and buffer.

Reply:
Ok, I moved the _createWorkDiv_ function as a new constructor. This PR could be merged, then I will prepare a second PR for the _array_ or _initilizer list_ type of inputs without explicit type parameters.

Reply:
It needed a deduction guide (hence a seperate constructor declaration with deduction guide) for some compilers.